Doc/library/exceptions.rst

```diff
@@ -866,6 +866,8 @@ The following exceptions are used as warning categories; see the
 
    Base class for warnings about dubious syntax.
 
+   This warning is emitted when compiling Python source code to bytecode.
```

Comment:
I think this is too strong - we probably don't want to commit to only using this exception type in the compilation process. You can write an application that raises these exceptions if you want.

Reply:
How about this?
By default, the Python interpreter emits the `SyntaxWarning` during bytecode compilation.

Reply:
That's not in the language spec either.

Reply:
I can't find a suitable word to describe the current behavior.

Reply:
I understand your concern that this document introduces some unofficially committed behavior for SyntaxWarning. I'm okay with leaving it here and waiting for more comments.

I searched the repository and found the following:

> SyntaxWarning are only emitted once, and only when a Python file is parsed: when a cached PYC file is used, not warning is emitted -------https://github.com/python/cpython/pull/99011#issuecomment-1301353061

> The problem is the warnings are only shown once, when the file is compiled.  
>
> Things would probably be improved if the warnings were always shown, as at least then devs will see the error once (although most will probably be confused when the warning doesn't repeat). --https://github.com/python/cpython/issues/77093#issuecomment-1093777501